omp/components/format_conversion.hpp

```diff
@@ -39,30 +39,69 @@ namespace kernels {
 namespace omp {
 
 
+/**
+ * @internal
+ *
+ * Converts an array of indexes `idxs` in any order to an array of pointers
+ * `ptrs`. This is used for transposing a csr matrix when calculating the row
+ * pointers of the transposed matrix out of the column indices of the original
+ * matrix.
+ */
 template <typename IndexType>
-inline void convert_idxs_to_ptrs(const IndexType *idxs, size_type num_nonzeros,
-                                 IndexType *ptrs, size_type length)
+inline void convert_unsorted_idxs_to_ptrs(const IndexType *idxs,
+                                          size_type num_nonzeros,
+                                          IndexType *ptrs, size_type length)
 {
-    std::fill(ptrs, ptrs + length, 0);
-    std::for_each(idxs, idxs + num_nonzeros, [&](size_type v) {
+#pragma omp parallel for schedule(static, \
+                                  ceildiv(length, omp_get_num_threads()))
```

Comment:
You actually need to include `omp.h` in order to get access to the function `omp_get_num_threads()`.

Reply:
A better idea would probably be to use `omp_get_max_threads()`, does the following:
> the omp_get_max_threads function returns an integer that is guaranteed to be at least as large as the number of threads that would be used to form a team if a parallel region without a num_threads clause were to be encountered at that point in the code.

--- taken from the [OpenMP documentation page 37](https://www.openmp.org/wp-content/uploads/cspec20_bars.pdf)

This would ensure that we always get the right number of threads instead of relying on the fact that we are already in a parallel region (from `omp parallel` in front of the `for`).